runtime/tests/interpreter/entitlements_test.go

```diff
@@ -577,6 +577,320 @@ func TestInterpretEntitledReferenceCasting(t *testing.T) {
 			value,
 		)
 	})
+
+	t.Run("capability downcast", func(t *testing.T) {
+
+		t.Parallel()
+
+		address := interpreter.NewUnmeteredAddressValueFromBytes([]byte{42})
+
+		inter, _ := testAccount(t,
+			address,
+			true,
+			`
+			entitlement X
+			entitlement Y
+
+			fun test(): Bool {
+				account.save(3, to: /storage/foo)
+				let capX = account.getCapability<auth(X, Y) &Int>(/public/foo)
+				let upCap = capX as Capability<auth(X) &Int>
+				return upCap as? Capability<auth(X, Y) &Int> == nil
+			}
+			`,
+			sema.Config{})
+
+		value, err := inter.Invoke("test")
+		require.NoError(t, err)
+
+		AssertValuesEqual(
+			t,
+			inter,
+			interpreter.TrueValue,
+			value,
+		)
+	})
+
+	t.Run("unparameterized capability downcast", func(t *testing.T) {
+
+		t.Parallel()
+
+		address := interpreter.NewUnmeteredAddressValueFromBytes([]byte{42})
+
+		inter, _ := testAccount(t,
+			address,
+			true,
+			`
+			entitlement X
+
+			fun test(): Bool {
+				account.save(3, to: /storage/foo)
+				let capX = account.getCapability<auth(X) &Int>(/public/foo)
+				let upCap = capX as Capability
+				return upCap as? Capability<auth(X) &Int> == nil
+			}
+			`,
+			sema.Config{})
+
+		value, err := inter.Invoke("test")
+		require.NoError(t, err)
+
+		AssertValuesEqual(
```

Comment:
Maybe return the resulting capability value and assert the value, instead of just that it is non-nil